mixer/tools/mixgen/cmd/adapter.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"bytes"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	gotemplate "text/template"
+
+	"github.com/spf13/cobra"
+
+	"istio.io/istio/mixer/cmd/shared"
+)
+
+const istioSystemNamespace = "istio-system"
+
+func adapterCfgCmd(rawArgs []string, printf, fatalf shared.FormatFn) *cobra.Command {
+	var resName string
+	var ns string
+	var configFilePath string
+	var description string
+	var sessionBased bool
+	var templates []string
+
+	adapterCmd := &cobra.Command{
+		Use:   "adapter",
+		Short: "creates kubernetes configuration for an adapter",
+		Run: func(cmd *cobra.Command, args []string) {
+			createAdapterCr("mixgen "+strings.Join(rawArgs, " "), resName, ns, description, configFilePath,
+				sessionBased, templates, printf, fatalf)
+		},
+	}
+	adapterCmd.PersistentFlags().StringVarP(&resName, "name", "n", "", "name of the resource")
+	adapterCmd.PersistentFlags().StringVar(&ns, "namespace", istioSystemNamespace,
+		fmt.Sprintf("namespace of the resource; default is %s", istioSystemNamespace))
```

Comment:
I don't think you need to have the default here, Cobra will already output that.